pandas/core/frame.py

```diff
@@ -8162,12 +8174,22 @@ def corr(self, method="pearson", min_periods=1) -> "DataFrame":
               dogs  cats
         dogs   1.0   0.3
         cats   0.3   1.0
+        >>> df.corr(method=histogram_intersection, axis=1)
+            0    1    2    3
+        0  1.0  0.3  0.2  0.3
+        1  0.3  1.0  0.0  0.1
+        2  0.2  0.0  1.0  0.2
+        3  0.3  0.1  0.2  1.0
         """
         numeric_df = self._get_numeric_data()
-        cols = numeric_df.columns
+        axis = numeric_df._get_axis_number(axis)
+        cols = numeric_df._get_agg_axis(axis)
         idx = cols.copy()
         mat = numeric_df.to_numpy(dtype=float, na_value=np.nan, copy=False)
 
+        if axis == 1:
```

Comment:
don't we have to transpose the results?

Reply:
I don't think we do since the result is symmetric